tools/apriltag/process_dynamics.py

```diff
@@ -0,0 +1,203 @@
+
+import csv
+import os
+import pdb
+import sys
+import time
+
+import numpy as np
+
+from scipy import signal
+from scipy.spatial.transform import Rotation, RotationSpline
+from scipy.interpolate import CubicSpline
+
+import matplotlib.pyplot as plt
+
+if len(sys.argv) < 3:
+    print("usage: python[3] process_dynamics.py [INBAG] [OUTCSV]")
+    sys.exit()
+
+# static parameters for toss detection
+TOSS_BALL_TOLERANCE = 1e-3
+TOSS_BALL_H = 10
+
+# far edge of board in board coordinates (y-axis)
+BOARD_EDGE_Y = 0.6
+
+# false if datapoint is over the board
+OFFBOARD_CONDITION = lambda data, i: data[i,1] > BOARD_EDGE_Y
+
+# true if datapoint is over the board
+ONBOARD_CONDITION = lambda data, i: not OFFBOARD_CONDITION(data, i)
+
+# true if block moves significantly over horizon H
+MOVING_CONDITION = lambda data, i: minBallOverHorizon(data[i:,:], TOSS_BALL_H) > TOSS_BALL_TOLERANCE
+
+# false if block moves significantly over horizon H
+STOPPED_CONDITION = lambda data, i: not MOVING_CONDITION(data, i)
+
+# returns minium ||\delta x|\ over a horizon of H in data
+def minBallOverHorizon(data, H):
+    diff_data = data[1:H,:] - data[0:(H-1),:]
+    return np.max(np.linalg.norm(diff_data,axis=1))
```

Comment:
Slightly confused why this returns the minimum Dx over a span of data. Shouldn't the return line read np.min?

Reply:
Alternatively, I feel like you might actually want the max movement based on the MOVING_CONDITION above, in which case maybe the function name should be adjusted

Reply:
renamed to maxDeltaOverHorizon